avm/res/storage/storage-account/tests/e2e/block/main.test.bicep

```diff
@@ -0,0 +1,48 @@
+targetScope = 'subscription'
+
+metadata name = 'Deploying as a Block Blob Storage'
+metadata description = 'This instance deploys the module as a Premium Block Blob Storage account.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-storage.storageaccounts-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
```

Comment:
@fblix will this work? I recall you mentioning that the block-storage can be an issue in places.

Reply:
yep this will fail if the region does not support premium type storage out of the box. Question is how we want to handle this in the future. Adding @eriqua @ChrisSidebotham to the conversation.

Reply:
Do we know if the default region we selected for the region rotation feature support this? ( eastus, uksouth, northeurope, eastasia ) 

If so we should be covered. If not, we should look into implementing exceptions

Reply:
@fblix & @krbar do you recall in which regions it is supported? I think last time we talked, only one of the regions in our rotation ('eastus', 'uksouth', 'northeurope', 'eastasia') supports block storage (could be mistaken though)

Reply:
Ok did some further digging. Seems the error message is misleading - its more a combination of kind and accesstier than anything else and I think @krbar 's nullable fix should also fix this issue.

Reply:
So it can be deployed in any region (or at least the most common ones)?